### PR TITLE
Audio: Peak volume: Add Kconfig to select the period of reporting peak meter value.

### PR DESCRIPTION
**Pull request by andrula-song**
Add kconfig to select the period of reporting peak meter value. The default value is reporting peak meter every 10 ms.

### CODE REVIEW

**Comment by andrula-song** [comment]
the perf of original code is:
[00:00:54.438,273] <inf> component: comp_copy: comp:0 0x60000 perf comp_copy samples 48 period 1000 cpu avg 333 peak 343
[00:00:54.437,185] <inf> component: comp_copy: comp:1 0x60001 perf comp_copy samples 48 period 1000 cpu avg 317 peak 325

the perf of setting as updating every 5ms is :
[00:00:40.105,926] <inf> component: comp_copy: comp:0 0x60000 perf comp_copy samples 48 period 1000 cpu avg 309 peak 346
[00:00:40.104,835] <inf> component: comp_copy: comp:1 0x60001 perf comp_copy samples 48 period 1000 cpu avg 301 peak 338

the perf of setting as updating every 10ms is :
[00:00:21.674,308] <inf> component: comp_copy: comp:0 0x60000 perf comp_copy samples 48 period 1000 cpu avg 301 peak 339

[00:00:22.697,213] <inf> component: comp_copy: comp:1 0x60001 perf comp_copy samples 48 period 1000 cpu avg 297 peak 336

the perf of setting as updating every 100ms is :
[00:00:24.744,386] <inf> component: comp_copy: comp:0 0x60000 perf comp_copy samples 48 period 1000 cpu avg 301 peak 333
[00:00:25.767,296] <inf> component: comp_copy: comp:1 0x60001 perf comp_copy samples 48 period 1000 cpu avg 297 peak 345

since the result of 10ms is almost the same with 100ms, so I set the default value as 10ms to update peak meter.  And in this way we can save about 9.6% cycles.

File: src/audio/module_adapter/module/volume/volume.c [hunk 1]
```diff
@@ -1169,6 +1170,14 @@ static int volume_process(struct processing_module *mod,
 
 		avail_frames -= frames;
 	}
+#if CONFIG_COMP_PEAK_VOL
+	cd->peak_cnt++;
```
Comment (btian1) [inline src/audio/module_adapter/module/volume/volume.c:1174]: should use frames to calculate time, we can't assume volume_process handle 1ms per time.

Comment (btian1) [reply src/audio/module_adapter/module/volume/volume.c:1174]: mod->dev->period/1000 is the ms

Comment (singalsu) [reply src/audio/module_adapter/module/volume/volume.c:1174]: I don't think this works with many instances and different scheduling. You need to calculate the number of copy() operations to per update for every peakvolume instance.

**Review by singalsu — CHANGES_REQUESTED** [review]

File: src/audio/module_adapter/module/volume/volume.c (hunk 1 above)
Comment (singalsu) [reply src/audio/module_adapter/module/volume/volume.c:1174]: So, please restore kconfig to ms instead of periods.

Comment (btian1) [reply src/audio/module_adapter/module/volume/volume.c:1174]: cd->peak_cnt += mod->dev->period/1000；
or
if (cd->peak_cnt == CONFIG_PEAK_METER_UPDATE_PERIOD*1000/mod->dev->period)

Comment (andrula-song) [reply src/audio/module_adapter/module/volume/volume.c:1174]: I'd like to move the count calculation to prepare function. Done thanks

**Review by btian1 — COMMENTED** [review]
before go to merge, shall we contact windows team to test peak_vol again? since current CI does not cover peak_vol case.

**Review by kv2019i — APPROVED** [review]

**Comment by kv2019i** [comment]
@btian1 wrote:
> before go to merge, shall we contact windows team to test peak_vol again? since current CI does not cover peak_vol case.

This is covered in "Internal Intel CI System/merge/codecheck" tests.

File: src/audio/module_adapter/Kconfig [hunk 2]
```diff
@@ -184,6 +184,42 @@ config COMP_PEAK_VOL
          This option enables reporting to host peak vol regs.
          See: struct ipc4_peak_volume_regs
 
+choice "PEAK_METER_UPDATE_PERIOD_CHOICE"
+	prompt "The periods(ms) of updating peak peter value"
```
Comment (ranj063) [inline src/audio/module_adapter/Kconfig:188]: Peak Peter is nice and alliterative but we should fix this typo I think , both here and the commit message

Comment (andrula-song) [reply src/audio/module_adapter/Kconfig]: thanks, done!

**Review by btian1 — APPROVED** [review]

**Comment by kv2019i** [comment]
#7191 is hitting in https://sof-ci.01.org/sofpr/PR7715/build9082/devicetest/index.html , but it's seen on the daily tests witout this PR. https://sof-ci.01.org/sofpr/PR7715/build9083/devicetest/index.html has only unrelated fails. I'll proceed with merge.

**Pull request merged**